core/src/consensus/stake/action_data.rs

```diff
@@ -257,6 +259,74 @@ impl IntermediateRewards {
     }
 }
 
+pub struct Candidates(BTreeMap<Address, Candidate>);
+#[derive(RlpEncodable, RlpDecodable, Clone)]
+#[cfg_attr(test, derive(Debug, Eq, PartialEq))]
```

Comment:
What is the reason that `Candidate` should derive these traits only in tests?

Reply:
No. I'll simplify them.